### PR TITLE
Add non-standard multi-value audio tag support

### LINKED ISSUE #10730: [Issue]: Multiple Artists Split Delimeter missing comma, or make it configurable
### Please describe your bug

In Music, Artists tag when contains multiple artists, most common practice is to add all by splitting with `, ` , which for some reason jellyfin doesnt split itself.

This has critical in having a usable artists tab in media-player clients, 
Pls add comma as a delimeter, or make delimeters configurable.

Ex: 
https://musicbrainz.org/doc/Style/Classical/Release_Artist

### Jellyfin Version

10.8.z

### if other:

_No response_

### Environment

```markdown
- OS: Ubuntu 20.04 LTS
- Linux Kernel: GNU/Linux 5.15.0-1049
- Virtualization: Docker
```

### Jellyfin logs

_No response_

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #10794: [Issue]: Treats artist names with slashes in them as multiple artists
### Please describe your bug

I recently added a rip of "[Black Science](https://www.discogs.com/master/122329-Geezer-Black-Science)," an album by "[G//Z/R](https://www.discogs.com/artist/11609-GZR)" (a side project of Geezer Butler).  The relevant FLAC tags for the files in this album are:

```
    comment[7]: ALBUM=Black Science
    comment[8]: ALBUM ARTIST=G//Z/R
    comment[9]: ALBUMARTIST=G//Z/R
    comment[10]: ALBUMARTIST_CREDIT=geezer
    comment[11]: ALBUMARTISTSORT=G//Z/R
    comment[16]: ARTIST=G//Z/R
    comment[17]: ARTIST_CREDIT=geezer
    comment[18]: ARTISTSORT=G//Z/R
    comment[24]: COMPILATION=0
```

The album is held in `/star/library/audio/flac/G__Z_R/Black Science`.

When Jellyfin shows the album in the album list, it says the artist is "G":

image

In the album detail, it says the artist is "G / R / Z":

image

I believe what JF is doing is treating this like an album by three artists, then sorting those names alphabetically for display, which is incorrect.  FLAC files which credit multiple artists will have multiple `ARTIST` (for multi-artist tracks) or `ALBUMARTIST` (for multi-artist albums) tags.

I'm not sure **why** it's doing this — maybe the double slashes is throwing it off?  Other artists I'd expect to see the same behavior with (ex. "AC/DC") are unaffected.

### Jellyfin Version

Other

### if other:

10.8.13

### Environment

```markdown
- OS: Debian 11
- Linux Kernel: Linux neutron 5.10.0-23-amd64 #1 SMP Debian 5.10.179-1 (2023-05-12) x86_64 GNU/Linux
- Virtualization: Docker
- Clients: Web
- Browser: [LibreWolf](https://librewolf.net)
- FFmpeg Version: n/a
- Playback Method: n/a
- Hardware Acceleration: n/a
- GPU Model:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
[2024-01-01 21:23:30.888 +00:00] [INF] [100] Emby.Server.Implementations.IO.LibraryMonitor: "Black Science" ("/media/audio/music/flac/G__Z_R/Black Science") will be refreshed.
[2024-01-01 21:28:08.386 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Man in a Suitcase" ("/media/audio/music/flac/G__Z_R/Black Science/01-Man in a Suitcase.flac") will be refreshed.
[2024-01-01 21:28:08.389 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Black Science" ("/media/audio/music/flac/G__Z_R/Black Science") will be refreshed.
[2024-01-01 21:28:08.389 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Box of Six" ("/media/audio/music/flac/G__Z_R/Black Science/02-Box of Six.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Mysterons" ("/media/audio/music/flac/G__Z_R/Black Science/03-Mysterons.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Justified" ("/media/audio/music/flac/G__Z_R/Black Science/04-Justified.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Department S" ("/media/audio/music/flac/G__Z_R/Black Science/05-Department S.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Area Code 51" ("/media/audio/music/flac/G__Z_R/Black Science/06-Area Code 51.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Has to Be" ("/media/audio/music/flac/G__Z_R/Black Science/07-Has to Be.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Number 5" ("/media/audio/music/flac/G__Z_R/Black Science/08-Number 5.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Among the Cybermen" ("/media/audio/music/flac/G__Z_R/Black Science/09-Among the Cybermen.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Unspeakable Elvis" ("/media/audio/music/flac/G__Z_R/Black Science/10-Unspeakable Elvis.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Xodiak" ("/media/audio/music/flac/G__Z_R/Black Science/11-Xodiak.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Northern Wisdom" ("/media/audio/music/flac/G__Z_R/Black Science/12-Northern Wisdom.flac") will be refreshed.
[2024-01-01 21:28:08.390 +00:00] [INF] [57] Emby.Server.Implementations.IO.LibraryMonitor: "Trinity Road" ("/media/audio/music/flac/G__Z_R/Black Science/13-Trinity Road.flac") will be refreshed.
[2024-01-01 21:28:40.096 +00:00] [INF] [26] Jellyfin.Plugin.KodiSyncQueue.EntryPoints.LibrarySyncNotification: Library Sync: Updated 9 items
[2024-01-01 21:28:40.096 +00:00] [INF] [26] Jellyfin.Plugin.KodiSyncQueue.EntryPoints.LibrarySyncNotification: Finished library sync, taking "00:00:06.3580063"
[2024-01-01 21:28:54.019 +00:00] [INF] [76] Jellyfin.Plugin.KodiSyncQueue.EntryPoints.LibrarySyncNotification: Started library sync
[2024-01-01 21:28:55.267 +00:00] [INF] [76] Jellyfin.Plugin.KodiSyncQueue.EntryPoints.LibrarySyncNotification: Library Sync: Updated 5 items
[2024-01-01 21:28:55.267 +00:00] [INF] [76] Jellyfin.Plugin.KodiSyncQueue.EntryPoints.LibrarySyncNotification: Finished library sync, taking "00:00:01.2475525"
```

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #11411: [Issue]: Jellyfin uses "Artist" tag rather than "Artists" tag
### Please describe your bug

So MusicBrainz Picard and therefore the entire MusicBrainz database uses "Artists" as the field for pulling the artists in the song and formats it with semicolons (when you set that as delimiter and use ID3V2.3) and uses "Artist" 

Look at screenshots to see more detail.

I would suggest using the "Artists" tag to determine which artists are in the song. Edit: Fallback to "Artist" if "Artists" is unpopulated.

Read about the difference which seems to stem from Kodi:
https://community.metabrainz.org/t/artist-vs-artists-tag/672616

### Reproduction Steps

Tag some music with multiple artists using MusicBrainz Picard. Add Music Library to Jellyfin with all the internet based metadata providers disabled except for artist related ones (so that you get artist images and descriptions, but only use embedded metadata for albums and songs).

### Jellyfin Version

Unstable (master branch)

### if other:

I tested on 10.8.13 as well as unstable.

### Environment

```markdown
- OS: TrueNAS Scale
- Linux Kernel:
- Virtualization: Docker
- Clients: Browser
- Browser: Chrome
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration:
- GPU Model:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
Not relevant as there is no error.
```

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

Jellyfin displaying artists:
Screenshot 2024-04-22 201642

Jellyfin Metadata not showing artists correctly:
Screenshot 2024-04-22 201256

Jellyfin making a mess by adding an artist for two artists but only for the times when the artists show up in "Artist" but for the other songs with other artists that aren't listed under "Artist" and only under "Artists" not showing up:
Screenshot 2024-04-22 202213

MusicBrainz Picard and the "Artist" and "Artists" tag:
Screenshot 2024-04-22 201323

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #11871: [Issue]: Music Artist Didn't Split by Semicolon
### Please describe your bug

I have a semicolon-separated artist "Adam Lambert;Nile Rodgers;Sam Sparro" in my ID3 metadata. However Jellyfin thinks there is only one artist called "Adam Lambert;Nile Rodgers;Sam Sparro". There should be 3 artists.

image

The more interesting thing is that if I click "Edit Metadata" on the web client, don't modify anything and click "Save Changes", the artist are correct now.

Here is the screen record:

https://github.com/jellyfin/jellyfin/assets/731326/e44e326e-c8ed-4cdb-be2c-b2b4edac3f8b

### Reproduction Steps

As above

### Jellyfin Version

10.9.0

### if other:

10.9.3

### Environment

```markdown
- OS: macOS 14.3.1
- Clients: Browser
- Browser: Chrome 122
- FFmpeg Version: 6.0.1
```

### Jellyfin logs

```shell
[2024-05-29 10:20:21.411 +08:00] [INF] [31] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Starting "/Applications/Jellyfin.app/Contents/MacOS/ffprobe" with args "-analyzeduration 200M -probesize 1G -i file:\"/mnt/media/MusicDLs/Adam Lambert;Nile Rodgers;Sam Sparro - Shady.mp3\" -threads 0 -v warning -print_format json -show_streams -show_format"
```

### FFmpeg logs

```shell
{
    "streams": [
        {
            "index": 0,
            "codec_name": "mp3",
            "codec_long_name": "MP3 (MPEG audio layer 3)",
            "codec_type": "audio",
            "codec_tag_string": "[0][0][0][0]",
            "codec_tag": "0x0000",
            "sample_fmt": "fltp",
            "sample_rate": "44100",
            "channels": 2,
            "channel_layout": "stereo",
            "bits_per_sample": 0,
            "initial_padding": 0,
            "r_frame_rate": "0/0",
            "avg_frame_rate": "0/0",
            "time_base": "1/14112000",
            "start_pts": 353600,
            "start_time": "0.025057",
            "duration_ts": 2525552640,
            "duration": "178.964898",
            "bit_rate": "261193",
            "disposition": {
                "default": 0,
                "dub": 0,
                "original": 0,
                "comment": 0,
                "lyrics": 0,
                "karaoke": 0,
                "forced": 0,
                "hearing_impaired": 0,
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 0,
                "timed_thumbnails": 0,
                "captions": 0,
                "descriptions": 0,
                "metadata": 0,
                "dependent": 0,
                "still_image": 0
            },
            "tags": {
                "encoder": "Lavc60.31"
            },
            "side_data_list": [
                {
                    "side_data_type": "Replay Gain"
                }
            ]
        },
        {
            "index": 1,
            "codec_name": "mjpeg",
            "codec_long_name": "Motion JPEG",
            "profile": "Baseline",
            "codec_type": "video",
            "codec_tag_string": "[0][0][0][0]",
            "codec_tag": "0x0000",
            "width": 300,
            "height": 300,
            "coded_width": 300,
            "coded_height": 300,
            "closed_captions": 0,
            "film_grain": 0,
            "has_b_frames": 0,
            "sample_aspect_ratio": "1:1",
            "display_aspect_ratio": "1:1",
            "pix_fmt": "yuvj444p",
            "level": -99,
            "color_range": "pc",
            "color_space": "bt470bg",
            "chroma_location": "center",
            "refs": 1,
            "r_frame_rate": "90000/1",
            "avg_frame_rate": "0/0",
            "time_base": "1/90000",
            "start_pts": 2255,
            "start_time": "0.025056",
            "duration_ts": 16106841,
            "duration": "178.964900",
            "bits_per_raw_sample": "8",
            "disposition": {
                "default": 0,
                "dub": 0,
                "original": 0,
                "comment": 0,
                "lyrics": 0,
                "karaoke": 0,
                "forced": 0,
                "hearing_impaired": 0,
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 1,
                "timed_thumbnails": 0,
                "captions": 0,
                "descriptions": 0,
                "metadata": 0,
                "dependent": 0,
                "still_image": 0
            },
            "tags": {
                "title": "0",
                "comment": "Cover (front)"
            }
        }
    ],
    "format": {
        "filename": "/mnt/media/MusicDLs/Adam Lambert;Nile Rodgers;Sam Sparro - Shady.mp3",
        "nb_streams": 2,
        "nb_programs": 0,
        "format_name": "mp3",
        "format_long_name": "MP2/3 (MPEG audio layer 2/3)",
        "start_time": "0.025056",
        "duration": "178.964898",
        "size": "5899088",
        "bit_rate": "263698",
        "probe_score": 51,
        "tags": {
            "title": "Shady",
            "artist": "Adam Lambert;Nile Rodgers;Sam Sparro",
            "album": "Trespassing (Explicit)",
            "TDOR": "2012",
            "encoder": "Lavf60.16.100",
            "REPLAYGAIN_TRACK_GAIN": "-6.51 dB",
            "REPLAYGAIN_TRACK_PEAK": "0.904022",
            "lyrics-eng": "[ti:Shady]\r\n[ar:Adam Lambert]\r\n[al:Trespassing]\r\n[by:]\r\n[offset:0]\r\n[00:05.53]Baby  I'm on the hunt\r\n[00:07.03]\r\n[00:07.78]Baby  I got my target on you\r\n[00:09.72]Trouble  that's what I want\r\n[00:10.97]\r\n[00:12.53]And I'm gonna do just what I have to\r\n[00:15.03]Get your a** down to the front\r\n[00:17.28]Go on and roll it out  I dare you\r\n[00:19.28]\r\n[00:19.97]Sorry I'm kinda drunk\r\n[00:21.28]\r\n[00:22.10]But did you just say your game was brand new \r\n[00:24.34]\r\n[00:25.22]No I ain't broken but I \r\n[00:26.04]\r\n[00:27.22]I need a fix\r\n[00:28.16]\r\n[00:29.60]That satisfaction when I \r\n[00:30.91]\r\n[00:31.91]I get my kicks\r\n[00:32.85]\r\n[00:33.85]I'm up all night\r\n[00:35.04]\r\n[00:36.54]I'm out of sight\r\n[00:37.79]\r\n[00:38.66]Don't turn on the light\r\n[00:39.85]\r\n[00:41.60]Cause I'm feeling so shady lately\r\n[00:43.04]\r\n[00:44.54]Come on and take me underground\r\n[00:46.02]\r\n[00:47.33]Deep under the street\r\n[00:49.33]Come on and take me  take me down\r\n[00:51.02]\r\n[00:52.21]The freaks like us can meet\r\n[00:53.26]\r\n[00:54.08]Turn it up turn it up turn it up turn it way up\r\n[00:57.08]Don't stop the beat\r\n[00:57.95]\r\n[00:58.83]Come on and turn me  turn me on\r\n[01:01.33]\r\n[01:02.13]Cause I'm feeling so shady lately\r\n[01:03.64]\r\n[01:09.02]Gonna go and get me a shot\r\n[01:11.39]Gonna knock it back forget all my blues\r\n[01:12.95]\r\n[01:13.88]Yeah that really hits the spot\r\n[01:15.14]\r\n[01:16.21]I'm feeling pretty fierce in my damn sin shoes\r\n[01:18.14]\r\n[01:19.02]No I ain't broken but I \r\n[01:20.39]\r\n[01:21.14]I need a fix\r\n[01:22.26]\r\n[01:23.57]That satisfaction when I \r\n[01:25.14]\r\n[01:25.95]I get my kicks\r\n[01:27.08]\r\n[01:27.77]I'm up all night\r\n[01:28.82]\r\n[01:30.58]I'm out of sight\r\n[01:31.52]\r\n[01:32.64]Don't turn on the light\r\n[01:33.83]\r\n[01:35.58]Cause I'm feeling so shady lately\r\n[01:37.01]\r\n[01:38.51]Come on and take me underground\r\n[01:40.52]\r\n[01:41.51]Deep under the street\r\n[01:42.52]\r\n[01:43.26]Come on and take me  take me down\r\n[01:45.51]\r\n[01:46.20]The freaks like us can meet\r\n[01:47.45]\r\n[01:48.07]Turn it up turn it up turn it up turn it way up\r\n[01:51.01]\r\n[01:51.70]Don't stop the beat\r\n[01:52.88]Come on and turn me  turn me on\r\n[01:55.20]\r\n[01:56.08]Cause I'm feeling so shady lately\r\n[01:57.58]\r\n[02:01.19]I throw more shade than a cloudy day\r\n[02:02.04]\r\n[02:05.23]The brakes don't work I'm just a runaway train\r\n[02:07.01]\r\n[02:09.83]Somebody come and help me out\r\n[02:11.38]\r\n[02:12.13]All I wanna be is free\r\n[02:13.45]\r\n[02:14.58]I was trying to get down now I can't out\r\n[02:18.07]I'm shady lately\r\n[02:19.31]Come on and take me underground\r\n[02:21.00]\r\n[02:22.44]Deep under the street\r\n[02:23.44]\r\n[02:24.07]Come on and take me  take me down\r\n[02:26.88]Freaks like us can meet\r\n[02:28.00]\r\n[02:28.88]Turn it up turn it up turn it up turn it way up\r\n[02:31.31]\r\n[02:32.13]Don't stop the beat\r\n[02:33.63]Come on and turn me  turn me out\r\n[02:35.88]\r\n[02:36.81]Cause I'm feeling so shady lately\r\n[02:38.92]\r\n[02:39.42]Shady lately  that's how I play\r\n[02:41.30]\r\n[02:43.62]They say maybe it's just a phase\r\n[02:46.99]\r\n[02:48.74]Shady lately  runaway train\r\n[02:51.68]\r\n[02:53.55]Blame the game  quit blaming the gays\r\n[02:56.36]Shady lately"
        }
    }
}
```

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #12256: [Issue]: Tracks with multiple artists
### Please describe your bug

I was using ";" to separate artists on tracks, for example in the "artist" field: Elvis Presley; Celine Dion

But since the last updates it no longer works and it takes it as a single artist

### Reproduction Steps

1. Add multiple artist with ";" to separate them
2. The two are recognized as one artist

### Jellyfin Version

10.9.0

### if other:

_No response_

### Environment

```markdown
- OS: Debian 12
- Linux Kernel: ?
- Virtualization: ?
- Clients: PC (jellyfin media player)
- Browser: Firefox
- FFmpeg Version: ?
- Playback Method: ?
- Hardware Acceleration: ?
- GPU Model: ?
- Plugins: ? 
- Reverse Proxy: ?
- Base URL: ?
- Networking: ? 
- Storage: ?
```

### Jellyfin logs

```shell
No logs because not a playback problem
```

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

Capture d’écran_2024-07-11_16-23-46

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gnattu**
This adds support for widely used non-standard methods of recording multiple values in audio tags. Notably:

- The non-standard ARTISTS tag used by Kodi and Picard
- The use of a custom character (e.g., `;`) as a delimiter for tag values.

A user-customizable whitelist to exclude certain items from tag splitting is also added, as some artists may have special characters in their names.

Enabling this feature is a per-library setting and is disabled by default.

**Changes**

- Adds customizable tag splitting in library options

**Issues**

Fixes #11871
Fixes #12256
Fixes #10730
Fixes #10794
Fixes #11411

### CODE REVIEW

**Review by Shadowghost — APPROVED** [review]
Wouldn't it make more sense to have the Whitelist globally?
We might also consider pre-filling it with our current Whitelist to not change behavior.

**Comment by gnattu** [comment]
> We might also consider pre-filling it with our current Whitelist to not change behavior.

We already changed the behavior in 10.9 so it does not matter. I prefer not to pre-fill it as this is very very user specific.

> Wouldn't it make more sense to have the Whitelist globally?

To have it per library gives more flexibility in my opinion, as we will never know what kind of weird names the artists want to call themselves.

**Comment by gnattu** [comment]
>  If 'A&B' is on the whitelist, what happens with 'A&B/C'? Does it split into 'A&B' and 'C', or into 'A', 'B', and 'C'?

Whitelisted item is always extracted first so it will be 'A&B' and 'C'

> Additionally, is it possible to use strings instead of single characters as delimiters? For example, the letter 'x' shouldn't be a delimiter, but ' x ' often is.

No you cannot due to how it is currently implemented and how the string is split, to do more complicated customization we will need something like custom regex but it is currently out of scope and I don't plan to add that currently.
 
>Regarding the whitelist, perhaps we could avoid splitting artists with the same content. For example, if 'A&B' is a single artist rather than two separate ones ('A' and 'B'), could we check their content first? If 'A' and 'B' share identical content (songs, albums, etc.), the program might automatically add 'A&B' to the whitelist.

This is not possible currently because each file is processed individually and probably in different threads as well, so cross-checking if they "share" same thing is a complicated logic and also out of scope.

Basically this feature is added as a compatibility feature so that you don't have to retag all your files. But if you really have some nasty files with very strange tagging you probably still need to retag them using the spec-defined way, but in this case you only need to retag the nasty files instead of every files.

**Comment by yuygfgg** [comment]
> >  If 'A&B' is on the whitelist, what happens with 'A&B/C'? Does it split into 'A&B' and 'C', or into 'A', 'B', and 'C'?
> 
> 
> 
> Whitelisted item is always extracted first so it will be 'A&B' and 'C'
> 
> 
> 
> > Additionally, is it possible to use strings instead of single characters as delimiters? For example, the letter 'x' shouldn't be a delimiter, but ' x ' often is.
> 
> 
> 
> No you cannot due to how it is currently implemented and how the string is split, to do more complicated customization we will need something like custom regex but it is currently out of scope and I don't plan to add that currently.
> 
>  
> 
> >Regarding the whitelist, perhaps we could avoid splitting artists with the same content. For example, if 'A&B' is a single artist rather than two separate ones ('A' and 'B'), could we check their content first? If 'A' and 'B' share identical content (songs, albums, etc.), the program might automatically add 'A&B' to the whitelist.
> 
> 
> 
> This is not possible currently because each file is processed individually and probably in different threads as well, so cross-checking if they "share" same thing is a complicated logic and also out of scope.
> 
> 
> 
> Basically this feature is added as a compatibility feature so that you don't have to retag all your files. But if you really have some nasty files with very strange tagging you probably still need to retag them using the spec-defined way, but in this case you only need to retag the nasty files instead of every files.
> 
> 
> 
> 

I see. Great thank to your fantastic work! As a compatibility feature, current implementation seems to cover over 90% of the situations and is quite enough.

**Review by crobibero — APPROVED** [review]

**Comment by hjkcai** [comment]
Thanks for your grate work! BTW How do I enable these options via some config file or anything without the need to recompile jellyfin?

**Comment by gnattu** [comment]
You need to wait for the web part to be merged in and wait for next week's unstable after that merge if you don't want to build yourself.

**Comment by NicholasFlamy** [comment]
So I was changing the `options.xml` on my test instance, and I was trying to (purposefully) get K-391 or Au/Ra to split just to test functionality. Can confirm that jellyfin unstable does have custom delimiter functionality! 👍

Notes for anyone messing with this:
```
  <UseCustomTagDelimiters>true</UseCustomTagDelimiters>
  <CustomTagDelimiters>
    <char>47</char>
    <char>124</char>
    <char>59</char>
    <char>92</char>
```
This will split Au/Ra into 2 artists.
If you change the 47 (/) to 45 (-) It won't split Au/Ra and it will split K-391.
Also, restart jellyfin after making a change to the `options.xml` so that it will apply.

**Comment by gnattu** [comment]
> This will split Au/Ra into 2 artists.

And the whitelist is exactly for this purpose, did you add whitelist for it?

**Comment by NicholasFlamy** [comment]
> > This will split Au/Ra into 2 artists.
> 
> And the whitelist is exactly for this purpose, did you add whitelist for it?

No, haven't tested that yet. I wanted to test having it split the artist. It did exactly what I expected. I will do more testing later, thank you for your work!

**Comment by OdinGitDat** [comment]
I would like to disable artist splitting completely, does this feature allow that?

**Comment by gnattu** [comment]
This is not even enabled by default. You need to explicitly enable it per library as the description of this PR

File: MediaBrowser.Model/Configuration/LibraryOptions.cs
Comment (nielsvanvelzen) [inline MediaBrowser.Model/Configuration/LibraryOptions.cs:120]: Late to the party but two issues here:

1. The default value causes an invalid OpenAPI spec - probably just remove it
2. The `char` type is invalid for OpenAPI, needs to be string

Comment (nielsvanvelzen) [reply MediaBrowser.Model/Configuration/LibraryOptions.cs:120]: We could add a custom filter for swashbuckle that maps the char type to string with min and max length set to 1. but that requires api generators to support that and interpret it as char.

Comment (gnattu) [reply MediaBrowser.Model/Configuration/LibraryOptions.cs:120]: That is a problem because we only want a single character for this option. This means server will need to have a private field to return the sanitized results, and the default value is an array of char so I'm not quite sure why that does not work.

Comment (nielsvanvelzen) [reply MediaBrowser.Model/Configuration/LibraryOptions.cs:120]: The array part is not a problem - that works fine. The issue with using char is that there is no char type in OpenAPI.

The more important issue here is the default value though, that one is preventing our TypeScript SDK from building as `null` (the value emitted in the OpenAPI schema) is not a valid default for an array.

**Comment by chunjiw** [comment]
Thank you! Cannot wait to try it out in 10.10!

I have a related question though, I'm wondering how people deals with "feat." in artist tag? You mentioned above that using string as delimiter is not supported, I understand. Do you have any suggestions about how to deal with "feat."?

**Comment by gnattu** [comment]
> Do you have any suggestions about how to deal with "feat."

You can do it the kodi way where record the one with things like `feat.` in standard ARTIST tag, and keep another machine-friendly tag separated by the delimiter you want in the non-standard ARTISTS tag, and enable prefer ARTISTS tag for this library. So probably you still need to re-tag those files.

**Comment by chunjiw** [comment]
Awesome! Sounds very doable. I don't mind re-tag some tracks every now and then. Thank you for your suggestion!

**Pull request merged**
